custom/icds/integrations/interfaces/phi.py

```diff
@@ -0,0 +1,91 @@
+from django.conf import settings
+
+from custom.icds.integrations.clients.phi import PHIApiClient
+from custom.icds.integrations.interfaces.exceptions.phi import (
+    AuthenticationFailure,
+    MissingCredentials,
+)
+
+
+class BasePHIInterface(object):
+    client_type = None
+
+    def __init__(self):
+        self._response = None
+
+    def _ensure_credentials(self):
+        if not self._api_key or not self._password:
+            raise MissingCredentials("Please set api key and password for phi")
```

Comment:
This method isn't called anywhere.  Why not move the API key and password (and the check) into the client?  It can raise an exception there, then this interface won't need to know about them.

Reply:
ah, missed to call that in `__init__`.

Ya i think having it in the interface just makes it fail earlier rather than waiting for the call through the client to happen.

Reply:
https://github.com/dimagi/commcare-hq/pull/25480/commits/44ca7cc2103fb37f8b6e40540d619cf5b21109cb

Reply:
I'm not convinced that distinction matters much - you can still do the check before any network requests.  The reason I'm belaboring this point is that I'd expect the division to be:
"client" code - interacts with the external endpoint, handles network stuff
"interface" code - presents a simple, minimal interface for other code which should only know about this API in an abstract sense.